### DISCUSSION TITLE
How to deploy and start Directus on Azure App Service?

### DISCUSSION THREAD

**Discussion by biapar** [directus/directus#7878] [Archive]
How to deploy and start Directus on Azure App Service?

**Comment by rijkvanzanten** [comment]
Azure App Service runs Docker images natively, so you can simply point it to the existing Docker image straight from DockerHub: https://hub.docker.com/r/directus/directus.

**Comment by matthewbellringer** [comment]
I've found it works nicely though a local install then a deploying that through VSCode's azure plugin.

1. Run the quickstart and get the app running locally
2. Open VSCode with the Azure plugin
3. Edit the package.json file - https://docs.directus.io/guides/installation/ubuntu/ - to add the start command
4. Copy your .env file up to the application settings (right click the "application settings" part of the app config VSCode, then choose "upload local settings", then choose the local .env file
5. Right click the Web App name you and choose "Deploy web app..."

**Reply by biapar** [reply]
Thanks. Could you provide an example of .env file? I tried, but I've an error on start.

**Reply by matthewbellringer** [reply]
It's the same .env file that Directus creates locally. If it works locally, it should work remotely! That's assuming, of course, that the database you're connecting to is accessible from Azure.

You should be able to find meaningful error logs with bit of digging. What does your error say?

**Comment by biapar** [comment]
I use sqlite.

**Reply by mcavaciocchi** [reply]
It cannot be done, since you need a path for the file and the files in your package are static.

**Reply by biapar** [reply]
So I've to use an external DB or a VPS

**Reply by rijkvanzanten** [reply]
Or you can use Azure Storage as the volume mount for Docker to store the database file on

**Reply by biapar** [reply]
Have you an example of the link to use sqllite with Azure storage? Thanks.

**Reply by rijkvanzanten** [reply]
We don't have an example for that for Directus specifically. https://docs.microsoft.com/en-us/azure/app-service/configure-connect-to-azure-storage?tabs=portal&pivots=container-linux goes into detail on how to setup Azure Storage as a shared volume in Container Apps. With that configured, you can use the `DB_FILENAME` to control where Directus looks for the Sqlite database
